shared/helpers/puppeteer.js

```diff
@@ -111,49 +125,94 @@ async function render(page, componentName, renderOptions, browserOptions) {
   // `evaluate`, only a `name` that maps to the error class (and not its `name`
   // property, which means we get a mangled value). As a workaround, we can
   // gather and `return` the values we need from inside the browser
-  const error = await page.evaluate(
-    async (selector, exportName, config) => {
-      const namespace = await import('govuk-frontend')
-
-      // Find all matching modules
-      const $modules = document.querySelectorAll(selector)
-
-      try {
-        // Loop and initialise all $modules or use default
-        // selector `null` return value when none found
-        ;($modules.length ? $modules : [null]).forEach(
-          ($module) => new namespace[exportName]($module, config)
-        )
-      } catch ({ name, message }) {
-        return { name, message }
-      }
-    },
-    selector,
-    exportName,
-    browserOptions?.config
-  )
-
-  // Throw Puppeteer errors back to Jest
-  if (error) {
-    throw new Error(
-      `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
-        `\n\t${error.name}: ${error.message}`,
-      { cause: error }
+  try {
+    const error = await page.evaluate(
+      async (selector, exportName, config) => {
+        const namespace = await import('govuk-frontend')
+
+        // Find all matching modules
+        const $modules = document.querySelectorAll(selector)
+
+        try {
+          // Loop and initialise all $modules or use default
+          // selector `null` return value when none found
+          ;($modules.length ? $modules : [null]).forEach(
+            ($module) => new namespace[exportName]($module, config)
+          )
+        } catch ({ name, message }) {
+          return { name, message }
+        }
+      },
+      selector,
+      exportName,
+      browserOptions?.config
     )
+
+    // Throw Puppeteer errors back to Jest
+    if (error) {
+      throw new Error(
+        `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
+          `\n\t${error.name}: ${error.message}`,
+        { cause: error }
+      )
+    }
+
+    await page.evaluateHandle('document.fonts.ready')
+  } finally {
+    // Disable middleware
+    page.off('request', middleware)
+
+    await page.setRequestInterception(false)
   }
 
   return page
 }
 
+/**
+ * Component preview HTTP request middleware
```

Comment:
**suggestion** I think it's worth describing what this method does in its docs. This can be infered from the code, but `middleware` may muddy things a little there (as well as leading with component preview, as the middleware deals with both serving the component preview and assets)

```suggestion
 * HTTP request middleware for component preview and assets
 *
 * Creates a handler for Puppeteer's request event that will:
 * - respond with the component preview if requested the provided `route`
 * - respond with the content for the relevant assets if requested an asset
 * - or let the request carry through
```

Alternatively, maybe renaming the function to `createRequestHandler` would help drop that blurb of doc:

```
/**
 * Create a handler for Puppeteer `request` event to serve component preview and assets
 * 
 * @param ...
 * @returns ...
 */
function createRequestHandler({ route, body, contentType = 'text/html' }) {
 // ...
}
```

Reply:
I think I'll split it out so it's only assets, we don't have to handle the preview here

Reply:
Done ✅